planning/mission_planning/mission_planner/lib/mission_planner_base.cpp

```diff
@@ -26,23 +26,27 @@
 
 namespace mission_planner
 {
-MissionPlanner::MissionPlanner() : pnh_("~"), tf_listener_(tf_buffer_)
+MissionPlanner::MissionPlanner(const std::string & node_name)
+: Node(node_name), tf_buffer_(get_clock()), tf_listener_(tf_buffer_)
 {
-  pnh_.param<std::string>("map_frame", map_frame_, "map");
-  pnh_.param<std::string>("base_link_frame", base_link_frame_, "base_link");
+  map_frame_ = declare_parameter("map_frame", "map");
+  base_link_frame_ = declare_parameter("base_link_frame", "base_link");
 
-  goal_subscriber_ = pnh_.subscribe("input/goal_pose", 10, &MissionPlanner::goalPoseCallback, this);
-  checkpoint_subscriber_ =
-    pnh_.subscribe("input/checkpoint", 10, &MissionPlanner::checkpointCallback, this);
+  using std::placeholders::_1;
 
-  route_publisher_ = pnh_.advertise<autoware_planning_msgs::Route>("output/route", 1, true);
+  goal_subscriber_ = create_subscription<geometry_msgs::msg::PoseStamped>(
+    "input/goal_pose", 10, std::bind(&MissionPlanner::goalPoseCallback, this, _1));
+  checkpoint_subscriber_ = create_subscription<geometry_msgs::msg::PoseStamped>(
+    "input/checkpoint", 10, std::bind(&MissionPlanner::checkpointCallback, this, _1));
+
+  route_publisher_ = create_publisher<autoware_planning_msgs::msg::Route>("output/route", 1);
   marker_publisher_ =
-    pnh_.advertise<visualization_msgs::MarkerArray>("debug/route_marker", 1, true);
+    create_publisher<visualization_msgs::msg::MarkerArray>("debug/route_marker", 1);
```

Comment:
If it is important that this keeps the latching behavior, this would need the transient_local durability QoS

Reply:
Thanks!
I've fixed the QoS of the publishers.

Reply:
:+1: